core/packages/contracts/src/OutboundChannel.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+import "openzeppelin/access/AccessControl.sol";
+import "./IOutboundChannel.sol";
+import "./IVault.sol";
+
+contract OutboundChannel is IOutboundChannel, AccessControl {
+    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+    bytes32 public constant SUBMIT_ROLE = keccak256("SUBMIT_ROLE");
+
+    mapping(bytes => uint64) public nonce;
+
+    IVault public vault;
+    uint256 public fee;
+
+    event Message(bytes dest, uint64 nonce, bytes payload);
+
+    event FeeUpdated(uint256 fee);
+
+    error FeePaymentToLow();
+
+    constructor(IVault _vault, uint256 _fee) {
+        _grantRole(ADMIN_ROLE, msg.sender);
+        _setRoleAdmin(SUBMIT_ROLE, ADMIN_ROLE);
+        vault = _vault;
+        fee = _fee;
+    }
+
+    function submit(
+        bytes calldata dest,
+        bytes calldata payload
+    ) external payable onlyRole(SUBMIT_ROLE) {
```

Comment:
not fully following how authentication works for this - looks like anyone with SUBMIT_ROLE can submit here.

Does the message also need to include msg.sender ?

How does, for example, the NativeTokens peer app on the substrate side verify/authorize that a message to it came from the NativeTokens smart contract?

Also, looks like this is a shared channel - I presume this is to-be-deprecated and replaced with a non-shared channel?